### ISSUE TITLE
model mismatch

### ISSUE THREAD

**Issue by LyraWang** [facebookresearch/mvfst-rl#11]
Excuse  me, I'm running the traffic_gen server and client using the cc_algo='rl', while loading traced_model.pt provided in mvfst-rl/models, but a size mismatch problem emerged. Could you tell me if I missed anything or should I set any params before testing? Thank you very much.

 _build/build/traffic_gen/traffic_gen --mode=client
i'm really using rl
I1111 16:54:13.475431 13896 ExampleClient.h:151] ExampleClient connecting to [::1]:6666
I1111 16:54:13.475687 13896 RLCongestionControllerFactory.h:34] Creating RLCongestionController
I1111 16:54:13.475867 13896 CongestionControlLocalEnv.cpp:21] Loading traced model from models/traced_model.pt
I1111 16:54:13.856813 13896 ExampleClient.h:81] ExampleClient connected to [::1]:6666
I1111 16:54:13.856878 13896 ExampleClient.h:178] ExampleClient wrote "hello", len=5 on stream=0
W1111 16:54:14.123860 13896 CongestionControlLocalEnv.cpp:36] onObservation: Still waiting for an update from model, skipping observation
terminate called after throwing an instance of 'std::runtime_error'
  what():  size mismatch, m1: [1 x 112], m2: [220 x 512] at ../../aten/src/TH/generic/THTensorMath.cpp:752
The above operation failed in interpreter, with the following stack trace:
at <string>:2:25

      def addmm(self: Tensor, mat1: Tensor, mat2: Tensor, beta: number = 1.0, alpha: number = 1.0):
          return self + mat1.mm(mat2)
                        ~~~~~~~ <--- HERE

      def batch_norm(input : Tensor, running_mean : Optional[Tensor], running_var : Optional[Tensor], training : bool, momentum : float, eps : float) -> Tensor:
          if training:
              norm_mean, norm_var = torch.batch_norm_update_stats(input, running_mean, running_var, momentum)
          else:
              norm_mean = torch._unwrap_optional(running_mean)
              norm_var = torch._unwrap_optional(running_var)
          norm_mean = torch._ncf_unsqueeze(norm_mean, input.dim())
          norm_var = torch._ncf_unsqueeze(norm_var, input.dim())

The above operation failed in interpreter, with the following stack trace:

*** Aborted at 1573462454 (unix time) try "date -d @1573462454" if you are using GNU date ***
PC: @     0x7fdd0a003e97 gsignal
*** SIGABRT (@0x3e800003647) received by PID 13895 (TID 0x7fdd0634a700) from PID 13895; stack trace: ***
    @     0x7fdd1855f890 (unknown)
    @     0x7fdd0a003e97 gsignal
    @     0x7fdd0a005801 abort
    @     0x7fdd0a9f8957 (unknown)
    @     0x7fdd0a9feab6 (unknown)
    @     0x7fdd0a9feaf1 std::terminate()
    @     0x7fdd0a9fed24 __cxa_throw
W1111 16:54:14.249354 13896 CongestionControlLocalEnv.cpp:36] onObservation: Still waiting for an update from model, skipping observation
    @     0x7fdd0dce76a0 torch::jit::InterpreterStateImpl::handleError()
    @     0x7fdd0dceeb4e torch::jit::InterpreterStateImpl::runImpl()
    @     0x7fdd0dce2cfc torch::jit::InterpreterState::run()
    @     0x7fdd0dcbf2f3 torch::jit::GraphExecutorImplBase::run()
    @     0x7fdd0dedc896 torch::jit::script::Method::operator()()
    @     0x560ecdc205c6 (unknown)
    @     0x7fdd0aa2966f (unknown)
    @     0x7fdd185546db start_thread
    @     0x7fdd0a0e688f clone

**Comment by viswanathgs** [comment]
Hi @LyraWang! Thanks for trying out mvfst-rl!

The error that you see is because more flags need to be passed to the `traffic_gen` binary such as the dimensions of the inputs to be passed to be model, etc. These flags are serialized into the `traced_model.flags.pkl` file in https://github.com/facebookresearch/mvfst-rl/tree/master/models. There are two ways of running this:

1. We have a wrapper script that already takes care of unpickling this file and calling it via pantheon. It can be easily run via the command in https://github.com/facebookresearch/mvfst-rl#evaluation but with `--mode=test_local`.
2. If you prefer to run `traffic_gen` binary directly, then unpickle `traced_model.flags.pkl` file. Then, in the resulting dict, for all keys that start with 'cc_env', pass these as CLI args to `traffic_gen` binary. Sample code to do this can be found at: https://github.com/viswanathgs/pantheon/blob/master/src/wrappers/mvfst_rl.py#L57.

Please let me know if you run into further issues!
